src/backend_telemetry_buffer.rs

```diff
@@ -0,0 +1,169 @@
+use pgx::*;
+
+#[pg_schema]
+mod _prom_ext {
+    use pgx::{prelude::*, *};
+    use std::{
+        mem::MaybeUninit,
+        sync::{Mutex, Once},
+    };
+
+    extension_sql!(
+        r#"
+        CREATE TYPE _prom_ext.backend_telemetry_rec AS (
+            time timestamp with time zone,
+            value BIGINT
+        );
+        "#,
+        name = "backend_telemetry_rec_decl"
+    );
+
+    const COMPOSITE_TYPE_NAME: &str = "_prom_ext.backend_telemetry_rec";
+
+    type BufferItem = composite_type!(COMPOSITE_TYPE_NAME);
+    struct Buffer {
+        mem_ctx: PgMemoryContexts,
+        list: PgList<BufferItem>,
```

Comment:
are we sure PgList is better than a Rust list type? Do we need a pg type here?

Reply:
I was trying to keep everything allocated inside the pg's memory contexts. It matters less with a long-lived ring buffer, but since list cells are dynamically allocated, I opted for something that would be 100% gone at the context reset.

Now I switched to a statically allocated ring buffer of Datum.